File: frontend/src/types/project.ts
import { File } from './file';
import { ActionBase, Action } from './action';

export interface ProjectSidebar {
    id: string;
    name: string;
}

export interface Project extends ProjectSidebar {
    file: File;
    actions: ActionBase[];
    active_action: Action | null;
    created_at: string;
}

File: backend/src/models/project.py
from pydantic import BaseModel
from datetime import datetime
from uuid import UUID
from .file import CreateFile, File
from typing import Optional, List
from .action import Action, ActionBase

class ProjectBase(BaseModel):
    id: UUID
    name: str

class ProjectCreate(BaseModel):
    name: str
    file_id: int
    file: CreateFile
    user_id: UUID

    class Config:
        from_attributes = True

class Project(ProjectBase):
    actions: List[ActionBase] = []
    active_action: Optional[Action] = None
    file: File
    created_at: datetime

    class Config:
        from_attributes = True
